### PR TITLE
Fix normalisation of multiline doc attributes

### PR DESCRIPTION
**Pull request by bash**
I noticed that multiline `#[doc = "..."]` attributes are not formatted properly when I ran [`cargo expand`](https://github.com/dtolnay/cargo-expand) on a proc_macro that emits doc attributes.

### CODE REVIEW

File: src/attr/doc_comment.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            CommentStyle::TripleSlash,
+        );
+    }
+
+    fn test_doc_comment_is_formatted_correctly(
+        literal: &str,
+        expected_comment: &str,
+        style: CommentStyle<'_>,
+    ) {
+        GLOBALS.set(&Globals::new(), || {
```
Comment (scampi) [inline src/attr/doc_comment.rs]: what are `Globals`/`GLOBALS` for ?

File: src/test/mod.rs [hunk 2]
```diff
@@ -313,6 +313,11 @@ fn self_tests() {
     );
 }
 
+fn is_nightly() -> bool {
```
Comment (scampi) [inline src/test/mod.rs]: there is a macro `is_nightly_channel` in `crate::config::config_type` already which can be reused.

File: src/test/mod.rs [hunk 3]
```diff
@@ -426,6 +431,16 @@ fn check_files(files: Vec<PathBuf>, opt_config: &Option<PathBuf>) -> (Vec<Format
     let mut reports = vec![];
 
     for file_name in files {
+        let sig_comments = read_significant_comments(&file_name);
```
Comment (scampi) [inline src/test/mod.rs]: It would be better if the changes on this file were done in a separate PR.

Comment (bash) [reply src/test/mod.rs]: See #3558

File: src/test/mod.rs (hunk 2 above)
Comment (bash) [reply src/test/mod.rs]: Addressed in 4745cec (in new PR)

File: src/attr/doc_comment.rs [hunk 4]
```diff
@@ -0,0 +1,92 @@
+use crate::comment::CommentStyle;
+use std::fmt::{self, Display};
+use syntax_pos::symbol::Symbol;
+
+pub(super) struct DocCommentFormatter<'a> {
```
[the remark this answers is not in the document]
Comment (bash) [reply src/attr/doc_comment.rs]: Addressed in 1ba35fc

File: src/attr/doc_comment.rs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+impl Display for DocCommentFormatter<'_> {
+    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let opener = self.style.opener().trim_end();
+
+        let literal_as_str = self.literal.as_str().get();
+        let line_count = literal_as_str.lines().count();
+        let last_line_index = line_count - 1;
+        let lines = literal_as_str.lines().enumerate();
```
[the remark this answers is not in the document]
Comment (bash) [reply src/attr/doc_comment.rs]: Ooh I totally forgot about peek 😆. Addressed in 8d4cb6e

File: src/attr/doc_comment.rs (hunk 1 above)
Comment (bash) [reply src/attr/doc_comment.rs]: It initialises the global interner required for `Symbol::gensym` afaik. No longer needed as I'm now passing a `&str` to the `DocCommentFormatter` (11a68d3)

File: tests/target/normalize_multiline_doc_attribute.rs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+// rustfmt-unstable: true
+// rustfmt-normalize_doc_attributes: true
+
+///This comment
```
Comment (topecongiro) [inline tests/target/normalize_multiline_doc_attribute.rs]: I think that we should add a leading space on each line.

Comment (bash) [reply tests/target/normalize_multiline_doc_attribute.rs]: There was a [comment](https://github.com/rust-lang/rustfmt/pull/3548/files#diff-4b9f3caeb7d15ba0422250fb31e55930L333) about this in the original code. Maybe I should add it again?

Comment (topecongiro) [reply tests/target/normalize_multiline_doc_attribute.rs]: Oh sorry. Please keep the current implementation, it is the correct behavior (cc https://doc.rust-lang.org/rustdoc/the-doc-attribute.html).

**Comment by topecongiro** [comment]
Thanks!
